### PR TITLE
Upd/cxx wrap v0.9

### PR DESCRIPTION
**Pull request by kalmarek**

### CODE REVIEW

File: src/call_function.jl [hunk 1]
```diff
@@ -1,53 +1,68 @@
...
+export @pm
+
+function CxxWrap.StdVector{CxxWrap.StdString}(vec::AbstractVector{<:AbstractString})
+    return CxxWrap.StdVector(convert.(CxxWrap.StdString, vec))
+end
 
 """
-    call_function(app::Symbol, func::Symbol, args...; void=false, kwargs...)
+    call_function([::Type{PropertyValue},] app::Symbol, func::Symbol, args...;
+        template_parameters=String[], unwrap=true, kwargs...)
```
Comment (benlorenz) [inline src/call_function.jl]:
```suggestion
        template_parameters=String[], kwargs...)
```
remove unwrap argument here as well

File: src/repl.jl [hunk 2]
```diff
@@ -29,9 +30,9 @@ end
 function LineEdit.complete_line(c::PolymakeCompletions, s)
    partial = REPL.beforecursor(LineEdit.buffer(s))
    full = LineEdit.input_string(s)
-   res = shell_complete(full)
-   offset = res[1]
-   proposals = res[2:end]
+   res = convert(Tuple{Int, String}, shell_complete(full))
```
Comment (benlorenz) [inline src/repl.jl]:
```suggestion
   res = convert(Tuple{Int, Array{String}}, shell_complete(full))
```

**Comment by saschatimme** [comment]
For the Mac issue, do we wait until this is resolved in CxxWrap or ist there a workaround?

File: src/matrices.jl [hunk 3]
```diff
@@ -9,37 +9,30 @@ function Matrix{T}(mat::AbstractMatrix) where T
...
 end
 
 #create Matrix from a sparse matrix
 function Matrix{T}(mat::AbstractSparseMatrix) where T <: VecOrMat_eltypes
-    r,c,v = SparseArrays.findnz(mat)
-    res = Matrix{T}(size(mat))
-    for i = 1:length(r)
-        res[r[i],c[i]] = v[i]
+    row, col, val = SparseArrays.findnz(mat)
+    res = Polymake.spzeros(T, size(mat)...)
```
Comment (saschatimme) [inline src/matrices.jl]: Does it make sense to have a proper zero matrix function?

File: src/perlobj.jl [hunk 4]
```diff
@@ -19,16 +19,20 @@ function bigobj(name::String, input_data::Pair{<:Union{Symbol,String}}...; kwarg
...
-    return take(obj, prop, convert(PolymakeType, val))
-end
+Base.propertynames(p::Polymake.BigObject) = Symbol.(complete_property(p, ""))
 
 function Base.setproperty!(obj::BigObject, prop::Symbol, val)
+    @assert prop != :cpp_object
     return take(obj, string(prop), convert(PolymakeType, val))
 end
 
+function Base.setproperty!(obj::BigObject, prop::Symbol, val::Ptr{Nothing})
```
Comment (saschatimme) [inline src/perlobj.jl]: Isn't `Ptr{Cvoid}` more idiomatic (in the C sense) than `Ptr{Nothing}` (I know they are the same on the julia side but still...)
```suggestion
function Base.setproperty!(obj::BigObject, prop::Symbol, val::Ptr{Cvoid})
```

**Comment by saschatimme** [comment]
This just disables the bad code path for now...

```diff
diff --git a/src/rationals.jl b/src/rationals.jl
index ab2bf3d..6393a58 100644
--- a/src/rationals.jl
+++ b/src/rationals.jl
@@ -1,9 +1,12 @@
-Rational(num::Int64, den::Int64) =
-    rational_si_si(convert(CxxLong, num), convert(CxxLong, den))
+# Disabled until https://github.com/JuliaInterop/CxxWrap.jl/issues/203 is resolved
+# Rational(num::Int64, den::Int64) =
+#     rational_si_si(convert(CxxLong, num), convert(CxxLong, den))
 
 function Rational(num::T, den::S) where {T<:Base.Integer, S<:Base.Integer}
     R = promote_type(promote_type(T, Int64), promote_type(S, Int64))
-    R == Int64 && return Rational(convert(Int64, num), convert(Int64, den))
+    # Workaround for https://github.com/JuliaInterop/CxxWrap.jl/issues/203
+    # Original R == Int64 && return Rational(convert(Int64, num), convert(Int64, den))
+    R == Int64 && return Rational(Integer(num), Integer(den))
     return Rational(Integer(convert(BigInt, num)), Integer(convert(BigInt, den)))
 end
 ```

**Comment by saschatimme** [comment]
Good to merge? :)

**Comment by kalmarek** [comment]
good to merge, when travis is green

**Pull request merged**